core/types/printing.cc

```diff
@@ -43,9 +43,33 @@ string UnresolvedAppliedType::toStringWithTabs(const GlobalState &gs, int tabs)
     return this->show(gs);
 }
 
+namespace {
+string argTypeForUnresolvedAppliedType(const GlobalState &gs, const TypePtr &t, ShowOptions options) {
+    if (auto *m = cast_type<MetaType>(t)) {
+        return m->wrapped.show(gs, options);
+    }
+    return t.show(gs, options);
+}
+} // namespace
+
 string UnresolvedAppliedType::show(const GlobalState &gs, ShowOptions options) const {
-    return fmt::format("{}[{}] (unresolved)", this->klass.show(gs, options),
-                       fmt::map_join(targs, ", ", [&](auto targ) { return targ.show(gs, options); }));
+    string resolvedString = options.showForRBI ? "" : " (unresolved)";
+    ClassOrModuleRef symForPrinting;
+
+    if (options.showForRBI) {
+        auto attachedClass = this->klass.data(gs)->attachedClass(gs);
+        symForPrinting = attachedClass;
+    } else {
+        symForPrinting = this->klass;
+    }
```

Comment:
I made similar changes to `UnresolvedClassType::show` when working on the initial single-package PR but rolled them back once stubbing was working correctly. Are you sure this change is needed? I would have assumed that types would resolve correctly after stubbing.

Reply:
I am "pretty" sure; without these changes, you'd get `UnresolvedAppliedType`s that print like `T.class_of(GenericMessage)[<Type: String>]` in the RBI, which doesn't seem like acceptable Ruby syntax.

It's possible that we should just change how `UnresolvedAppliedType` is constructed so we would have to do less of this.

Reply:
Maybe we could do some follow-up investigation to see about removing this special case; it seems like the sort of thing we should be able to avoid if stubbed constants look appropriately generic.